### PR TITLE
Switch `gspread` from `Client` to `BackoffClient`

### PR DESCRIPTION
**Pull request by jlumbroso**
This will solve errors of this type:
```
Traceback (most recent call last):
  File "/Users/jlumbroso/.local/share/virtualenvs/argos-pilot-XZjiRVec/bin/roseingrave", line 8, in <module>
    sys.exit(cli())
  File "/Users/jlumbroso/.local/share/virtualenvs/argos-pilot-XZjiRVec/lib/python3.9/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/Users/jlumbroso/.local/share/virtualenvs/argos-pilot-XZjiRVec/lib/python3.9/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/Users/jlumbroso/.local/share/virtualenvs/argos-pilot-XZjiRVec/lib/python3.9/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/jlumbroso/.local/share/virtualenvs/argos-pilot-XZjiRVec/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/jlumbroso/.local/share/virtualenvs/argos-pilot-XZjiRVec/lib/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/Users/jlumbroso/.local/share/virtualenvs/argos-pilot-XZjiRVec/lib/python3.9/site-packages/roseingrave/volunteer_summary.py", line 147, in volunteer_summary
    success, volunteer_data = export_spreadsheet(
  File "/Users/jlumbroso/.local/share/virtualenvs/argos-pilot-XZjiRVec/lib/python3.9/site-packages/roseingrave/volunteer_summary.py", line 55, in export_spreadsheet
    success, piece_data = Piece.export_sheet(sheet, template)
  File "/Users/jlumbroso/.local/share/virtualenvs/argos-pilot-XZjiRVec/lib/python3.9/site-packages/roseingrave/_piece.py", line 589, in export_sheet
    ) = _export_helper(sheet, template, is_master=True)
  File "/Users/jlumbroso/.local/share/virtualenvs/argos-pilot-XZjiRVec/lib/python3.9/site-packages/roseingrave/_piece.py", line 1086, in _export_helper
    values = sheet.get_values(value_render_option='formula')
  File "/Users/jlumbroso/.local/share/virtualenvs/argos-pilot-XZjiRVec/lib/python3.9/site-packages/gspread/utils.py", line 634, in wrapper
    return f(*args, **kwargs)
  File "/Users/jlumbroso/.local/share/virtualenvs/argos-pilot-XZjiRVec/lib/python3.9/site-packages/gspread/worksheet.py", line 396, in get_values
    return fill_gaps(self.get(range_name, **kwargs))
  File "/Users/jlumbroso/.local/share/virtualenvs/argos-pilot-XZjiRVec/lib/python3.9/site-packages/gspread/utils.py", line 634, in wrapper
    return f(*args, **kwargs)
  File "/Users/jlumbroso/.local/share/virtualenvs/argos-pilot-XZjiRVec/lib/python3.9/site-packages/gspread/worksheet.py", line 706, in get
    response = self.spreadsheet.values_get(range_name, params=params)
  File "/Users/jlumbroso/.local/share/virtualenvs/argos-pilot-XZjiRVec/lib/python3.9/site-packages/gspread/spreadsheet.py", line 181, in values_get
    r = self.client.request("get", url, params=params)
  File "/Users/jlumbroso/.local/share/virtualenvs/argos-pilot-XZjiRVec/lib/python3.9/site-packages/gspread/client.py", line 86, in request
    raise APIError(response)
gspread.exceptions.APIError: {'code': 429, 'message': "Quota exceeded for quota metric 'Read requests' and limit 'Read requests per minute per user' of service 'sheets.googleapis.com' for consumer 'project_number:576121055363'.", 'status': 'RESOURCE_EXHAUSTED', 'details': [{'@type': 'type.googleapis.com/google.rpc.ErrorInfo', 'reason': 'RATE_LIMIT_EXCEEDED', 'domain': 'googleapis.com', 'metadata': {'service': 'sheets.googleapis.com', 'consumer': 'projects/576121055363', 'quota_limit_value': '60', 'quota_limit': 'ReadRequestsPerMinutePerUser', 'quota_location': 'global', 'quota_metric': 'sheets.googleapis.com/read_requests'}}, {'@type': 'type.googleapis.com/google.rpc.Help', 'links': [{'description': 'Request a higher quota limit.', 'url': 'https://cloud.google.com/docs/quota#requesting_higher_quota'}]}]}
```
The `BackoffClient` uses exponential backoff to wait when it gets a quota exceeded error. This switch is completely invisible and will not do anything but ensure this error never happens!

I also bumped the required version of the library `gspread` to 5.5.0 since that is the version with which this change (which I suggested) was introduced:
https://github.com/burnash/gspread/blob/9c83ec0fe6499c30534c1f8d92919acaaa09a81c/HISTORY.rst#550-2022-08-31

### CODE REVIEW

**Comment by jlumbroso** [comment]
@josephlou5 Once you've accepted this, you can bump the version number and publish this package.

File: src/roseingrave/_sheets.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
 # ======================================================================
 
 import google.auth.exceptions
+import gspreed.client
```
Comment (josephlou5) [inline src/roseingrave/_sheets.py]: Could you replace these import statements (both `import gspread.*` statements) with `import gspread` instead? I believe that should cover everything.

File: src/roseingrave/_sheets.py [hunk 2]
```diff
@@ -70,6 +71,7 @@ def _error(msg):
         'credentials_filename': str(filepath),
         'authorized_user_filename': str(auth_user_path),
         'flow': gspread.auth.console_flow,
+        'client': gspreed.client.BackoffClient,
```
Comment (josephlou5) [inline src/roseingrave/_sheets.py]: This should be `'client_factory': gspread.client.BackoffClient`. Note the typo in `gspread`.

**Review by josephlou5 — CHANGES_REQUESTED** [review]
Do you also need to change the gspread version in `poetry.lock`?

File: src/roseingrave/_sheets.py (hunk 2 above)
Comment (jlumbroso) [reply src/roseingrave/_sheets.py]: Good catch, sorry about the sloppiness! 😅 
https://docs.gspread.org/en/latest/api/top-level.html#gspread.oauth

**Comment by jlumbroso** [comment]
@josephlou5 Made all the changes your requested, including to `poetry.lock`

If you have issues, you can refresh the lock file on your end too:
```
poetry update --lock
```

**Review by josephlou5 — APPROVED** [review]

**Pull request merged**